workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/hooks/useAutoSave.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {useEffect} from 'react';
+
+export default function useAutoSave(
+	autoSaveFunction: Function,
+	autoSaveTime: number,
+	monitoredObject: object,
+	shouldMonitor: boolean
```

Comment:
maybe instead of shouldMonitor we can call it 'clear' and if its true we clear the interval? that way it could handle both the dont use and the stop using use case?